applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -88,20 +101,14 @@ struct dvfs_state_timeout {
 	uint16_t timeout_ms;
 };
 
+static struct k_work dvfs_notify_work;
+static uint32_t requested_freq;
```

Comment:
Yes we probably could. I just would like to remind you that you were the one who claimed that we cannot add test_state to states because it behaves differently then rest of states. Those states you suggest would also be exceptions and work differently to other states. If you think it is a good idea I suggest doing this change together with adding test_state.

Reply:
I was against introducing a `DVFS_STATE_TEST` bit to the `dfvs_requests_state_bitmask` as it would work in another way than bits from other modules (I am still against it). Introducing a separate test state for whole module sounds reasonable.

Reply:
If you think expanding state machine is a good idea (as it would allow us for better state checks) then please open a follow-up ticket to ensure we will remember about it.

Reply:
Hmm, I thought you want to place it in dvfs_state. I see those states as states which can coexists with each other:

STATE_DISABLED, STATE_READY, STATE_ERROR - are exclusive while STATE_WAITING_FOR_DVFS_NOTIFY and STATE_RETRY_SCHEDULED are not. We can be at the same time in STATE_ERROR (we encountered error), STATE_WAITING_FOR_DVFS_NOTIFY (still waiting for some callback)  and STATE_RETRY_SCHEDULED (retry is not cancelled). I know that we could do some gimmick to make sure that all states are exclusive but I feel it would be artificial. IMO much better place for that would be in  dvfs_state.

Reply:
The `STATE_WAITING_FOR_DVFS_NOTIFY ` and `STATE_RETRY_SCHEDULED ` would be mutually exclusive if you consider `STATE_READY` as a state where proper frequency is already applied. After you enter `STATE_ERROR` you should no longer wait for DVFS notify nor retry. It does not sound artificial to me.

Looking at the current implementation, introducing it to `dvfs_state` would definitely not fit (those bits would once again be handled completely differently than other bits which describe tracked application state).

Reply:
IMO you will change from STATE_WAITING_FOR_DVFS_NOTIFY to change error before notification will happened and will loose information about fact that you are waiting for this notification.

Lets stop here and add link to this in task related to TEST_STATE.

Reply:
Added to task: https://nordicsemi.atlassian.net/browse/NCSDK-28467

Reply:
In error state you could actually ignore the notification, so it should not be a problem